src/lib/genesis_constants/genesis_constants.ml

```diff
@@ -191,7 +191,8 @@ module T = struct
           [t.protocol.k; t.protocol.delta; t.txpool_max_size]
           ~f:Int.to_string
       |> String.concat ~sep:"" )
-      ^ Core.Time.to_string t.protocol.genesis_state_timestamp
+      ^ Core.Time.to_string_abs ~zone:Time.Zone.utc
```

Comment:
It still feels safer to me to use millis since 1970 this will work for any behavior of time library  — I ran into a bug once because I’m Germany months years and days are separated by dots instead of dashes. Is this locals detail also coupled to core time’s api? I can’t easily find the docs for core time to confirm string_abs is safe. Anyway, I’ll assume you did if you land it as is (I won’t block on this)

Reply:
Sad :( there doesn't seem to be a good way to do this with the Core.Time API -- at least this `to_string_abs` API has: "This string can be round-tripped, even on a machine in a different time zone than the machine that wrote the string." So it's probably okay.